crates/pathfinder/src/monitoring/metrics.rs

```diff
@@ -0,0 +1,80 @@
+pub mod middleware {
+    use jsonrpsee::core::middleware::Middleware;
+    use metrics::increment_counter;
+    use std::time::Instant;
+
+    #[derive(Debug, Clone)]
+    pub struct RpcMetricsMiddleware;
+
+    impl Middleware for RpcMetricsMiddleware {
+        type Instant = Instant;
+
+        fn on_request(&self) -> Self::Instant {
+            Instant::now()
```

Comment:
This would work with `()` and `type Instant = ();` or is there something I'm not understanding?

Reply:
I added `std::time::Instant` anticipating that we would measure the times elapsed later, never mind. It's yagni right now.

Reply:
Yeah this was more like question to make sure I understood correctly that there are no time measurements currently! Resolving.